cpp/algo_module/algorithm/algo.cpp

```diff
@@ -1,5 +1,132 @@
 #include "algo.hpp"
 
+#include "mgp.hpp"
+
+Algo::PathFinder::PathFinder(const mgp::Node &start_node, const mgp::Node &end_node, int64_t max_length,
+                             const mgp::List &rel_types, const mgp::RecordFactory &record_factory)
+    : start_node_(start_node), end_node_id_(end_node.Id()), max_length_(max_length), record_factory_(record_factory) {
+  UpdateRelationshipDirection(rel_types);
+}
+
+void Algo::PathFinder::UpdateRelationshipDirection(const mgp::List &relationship_types) {
+  if (relationship_types.Size() == 0) {  // if no relationships were passed as arguments, all relationships are allowed
+    any_outgoing_ = true;
+    any_incoming_ = true;
+    return;
+  }
+
+  for (const auto &rel : relationship_types) {
+    std::string rel_type{std::string(rel.ValueString())};
+    bool starts_with = rel_type.starts_with('<');
+    bool ends_with = rel_type.ends_with('>');
+
+    if (rel_type.size() == 1) {
```

Comment:
Can you comment here if `<` or `>` is sent, then for all relationships `any_incoming_` or `any_outgoing_` works